FEAT - Split Lab build into an action of its own (#30)
Quansight-Labs/jupyter-a11y-testing: README.md

After JupyterLab, though it is not within the scope of the grant driving the roadmap,
we hope to extend this testing to other parts of the Jupyter ecosystem beyond JupyterLab.

## Repository orientation 🗺️

This repository is organized as follows:

```txt
.
├── .github
│   ├── .github/workflows                               # set of GitHub actions to run the accessibility tests based on certain type of triggers
│   └── .github/actions                                 # composable actions that perform specific tasks (not to be used on their own but as part of a GitHub actions workflow)
├── testing                                             # root folder for the testing tools
│   └── testing/jupyterlab                              # testing tools and scripts for JupyterLab

> │       ├── testing/jupyterlab/manual-testing-scripts   # "recipes" to perform manual accessibility checks on JupterLab (to be used in conjunction with the automated tests)

Reviewer: ```suggestion
│       ├── testing/jupyterlab/manual-testing-scripts   # "recipes" that explain in plain language how automated tests can be also be carried out manually
```

Reviewer: This makes me think there's something wrong with the naming or information hierarchy. This is why I originally named the directory something like test-info-pages... In my mental model, every accessibility test in this repo should come with its own human-readable page that explains what the test is, why it matters, how to carry out the test manually, and maybe other info.

Author: yes we need to discuss this on Monday I once again was not sure when creating the tree for the repo - will leave it as is now and we can make a decision on Monday

│       ├── testing/jupyterlab/tests                    # set of Playwright automated tests
│       ├── testing/jupyterlab/environment.yml          # conda environment file to install the dependencies for the automated tests
│       ├── testing/jupyterlab/README.md                # documentation for the JupyterLab tests - start here to learn how to run the tests locally or in GitHub actions
│       ├── testing/jupyterlab/package.json             # npm package file to install the dependencies for the automated tests
│       └── testing/jupyterlab/playwright.config.ts     # Playwright configuration file

> │   └── testing/notebooks                               # set of reference Jupter notebooks to be used in the automated tests

Reviewer: ```suggestion
│   └── testing/notebooks                               # set of reference Jupyter notebooks to be used in the automated tests
```

> │   └── testing/scripts                               # set of manual testing scripts for JupyterLab (include relevant WCAG success criteria and step-by-step guides to audit)

Reviewer: ```suggestion
│   └── testing/scripts                                 # set of manual testing scripts for JupyterLab (include relevant WCAG success criteria and step-by-step guides to audit)
```

Reviewer: We should probably delete `testing/scripts` lest it cause confusion with `testing/jupyterlab/manual-testing-scripts`.

Or we need to decide on the path forward for this repo and our work.

> ├── .pre-comMit-config.yaml                             # configuration file for the pre-commit hooks

Reviewer: ```suggestion
├── .pre-commit-config.yaml                             # configuration file for the pre-commit hooks
```
